### PR TITLE
Added information on xrefs for overloads, generic types to template

### PR DESCRIPTION
**Pull request by rpetrusha**
## Added information on xrefs for overloads, generic types to template

See #7857 

@imba-tjd

### CODE REVIEW

File: styleguide/template.md [hunk 1]
```diff
@@ -128,17 +128,24 @@ If a URL appears in a Markdown file, it will be transformed into a clickable lin
...
+
+To link to a specific method overload, you must include the fully qualified type name of each of the method's parameters. For example, \<xref:System.DateTime.ToString> links to the parameterless [DateTime.ToString](https://docs.microsoft.com/dotnet/api/system.datetime.tostring#System_DateTime_ToString) method, while \<xref:System.DateTime.ToString(System.String,System.IFormatProvider)> links to the  [DateTime.ToString(String,IFormatProvider)](https://docs.microsoft.com/dotnet/api/system.datetime.tostring#System_DateTime_ToString_System_String_System_IFormatProvider_) method.
+
+To link to a generic type, such as [System.Collections.Generic.List\<T>](https://docs.microsoft.com/dotnet/api/system.collections.generic.list-1), you use the ` (%60) character followed by the number of generic type parameters. For example, \<xref:System.Nullable%601> links to the [System.Nullable\<T>](https://docs.microsoft.com/dotnet/api/system.nullable-1) type, while \<xref:System.Func%602> links to the [System.Func\<T,TResult>](https://docs.microsoft.com/dotnet/api/system.func-2) delegate.
```
Comment (imba-tjd) [inline styleguide/template.md]: Hi. Would ``you use a ` (or %60) character`` be better?

Comment (nemrism) [reply styleguide/template.md]: Hi @imba-tjd,
I had to get through the same question recently and found the answer here:
https://github.com/dotnet/docs/blob/master/styleguide/template.md#links-to-apis

> When the UID contains the special characters `, # or *, the UID value needs to be HTML encoded as %60, %23 and %2A respectively. You'll sometimes see parentheses encoded but it's not a requirement.
> 
> Examples:
> 
> * System.Threading.Tasks.Task`1 becomes System.Threading.Tasks.Task%601
> * System.Exception.#ctor becomes System.Exception.%23ctor
> * System.Lazy`1.#ctor(System.Threading.LazyThreadSafetyMode) becomes System.Lazy%601.%23ctor%28System.Threading.LazyThreadSafetyMode%29

Comment (nemrism) [reply styleguide/template.md]: Not sure this is exactly what you expected as an answer, but I hope it helps

Comment (imba-tjd) [reply styleguide/template.md]: Oh... you are right. There is a ctor with parameter example. I can understand it now but it doesn't explicitly point out how to use parameter and I missed it when I first looked up it.

**Review by BillWagner — APPROVED** [review]
These are great clarifications @rpetrusha 

You can :shipit: when ready.

**Comment by rpetrusha** [comment]
I've updated the topic to add the ref service, if you'd like to take a look, @BillWagner.

**Pull request merged**